Use bazel to create boskos images

sebastienvas opened the pull request:
I think this was a long overdue. I am currently working on Boskos and decided to start with that to ease development.

sebastienvas commented:
@Katharine do I need to change the cloudbuild.yaml ?

krzyzacy commented:
glad to see this happen! punt to image experts:

/assign @Katharine @fejta

Katharine commented:
> @Katharine do I need to change the cloudbuild.yaml ?

You should probably just remove them, as well as `boskos/variants.yaml` (though before doing so, also consider what to do with `aws-janitor`, which you have not touched here).

You'll also want to remove the [`post-test-infra-push-boskos` job](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L101-L128).

sebastienvas commented:
Added aws image but did not delete the dockerfile as I don't know how it is used by Amazon team. Also updated the prow jobs by basically copying the one from prow. @fejta could you please check I did it right ?

fejta commented on boskos/BUILD.bazel:
```diff
@@ -1,5 +1,8 @@
 package(default_visibility = ["//visibility:public"])
 
+load("@io_bazel_rules_docker//go:image.bzl", "go_image")
```
Can we use prow_image so this has a good publish date?

sebastienvas replied:
done.

fejta commented on boskos/BUILD.bazel:
```diff
@@ -13,6 +16,27 @@ go_binary(
+)
+
+prow_push(
+    name = "release-push",
```
maybe just push?

sebastienvas replied:
done.

fejta commented on boskos/push.sh:
```diff
@@ -0,0 +1,77 @@
+set -o nounset
+set -o pipefail
+
+# TODO(fejta): rewrite this in a better language REAL SOON
```
Might be good to have a TODO to make this script reusable

sebastienvas replied:
done.

fejta commented on boskos/BUILD.bazel:
```diff
@@ -13,6 +16,27 @@ go_binary(
     pure = "on",
 )
 
+go_image(
```
This will use a distroless image by default, is that okay?

sebastienvas replied:
yes that s fine.

sebastienvas commented:
@fejta PTAL

fejta commented on maintenance/aws-janitor/BUILD.bazel:
```diff
@@ -23,6 +24,16 @@ go_binary(
     visibility = ["//visibility:public"],
 )
 
+go_image(
```
prow_image for this one too
